.goreleaser.yml

```diff
@@ -76,14 +76,16 @@ checksum:
 signs:
   -
     id: with_key_id
-    # TODO: Replace with variable once goreleaser supports variables here
+    # TODO: Replace with variable once signore provides the primary_key_id
```

Comment:
This is a good idea in principle but I assume that we'd have to obtain the key ID in a separate step and _then_ pass it down as ENV variable to goreleaser, right?

Presumably we'd want to be sure that the key obtained is the same key actually used for signing - so I suppose signore would then need to accept the key ID as a flag and reject signing if it doesn't match or something - just to prevent a rare case where the active key is changed between the two steps and we'd then end up publishing `sig` file under key ID which doesn't reflect the real key used.

Reply:
Yes, I would obtain the id separately and pass it down as an ENV variable to goreleaser.

I assume that obtaining the ID would work by using the `signore` binary together with the `SIGNORE_SIGNER` secret, like `signore get_signer $SIGNORE_SIGNER`. The returned object will have metadata containing a `primary_key_id` (the primary key) and a `key_id` (the subkey used for signing). I understand that there could be a rare case that the signer gets updated after obtaining the key id and before signing the binary. But I think it is much more likely having the id hardcoded here and forgetting to change it.

A way to avoid this case would be to have signore return some metadata with the `sign` call. Then, we can rename the `.sig` file with the correct key id from the metadata.

Reply:
> Then, we can rename the .sig file with the correct key id from the metadata.

That could be a little tricky because AFAIK GoReleaser uses the pre-declared `signature` under `signs` as a way of tracking the artifacts which are then passed e.g. to `hc-releases` later.

However I was merely curious about the context behind the comment at this point. We can cross the bridge when we come to it. 🤷🏻‍♂️ 😄